torchtune/models/llama3/_component_builders.py

```diff
@@ -250,8 +257,20 @@ def lora_llama3(
     if quantize_base:
         # For QLoRA, we reparametrize 4-bit tensors to bf16, and offload to CPU on the fly
         # so as to not increase peak memory
+        if low_cpu_ram and _low_ram_reparametrize_as_dtype_state_dict_post_hook is None:
+            if sys.platform == "win32":
+                raise RuntimeError(
+                    "low_cpu_ram=True not supported on Windows."
+                )
+            else:
+                raise RuntimeError("low_cpu_ram=True requires torch.__version__ >= 2.5.0.dev20240830.")
```

Comment:
Similar comment here: ideally we can do these checks in the recipe (or in a utility) rather than in the builder of the model